src/components/SidebarSearch/index.tsx

```diff
@@ -1,38 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FaSearch } from 'react-icons/fa/index';
 import { MdClear } from 'react-icons/md/index';
-import { AiOutlineRight } from 'react-icons/ai/index';
 import * as Styles from './styles';
 
-const SidebarSearch: React.FC = () => {
+import SearchItem from '../SearchItem/index';
+
+import { requestByName } from '../../services/api.requests';
+
+interface SidebarSearchProps {
+	updateState(status: boolean): void;
+}
+
+const SidebarSearch: React.FC<SidebarSearchProps> = ({ updateState }) => {
+	const [searchInput, setSearchInput] = useState<string>('');
+	const [results, setResults] = useState<Record<any, any>[]>([]);
+
+	function handleClose(): void {
+		setSearchInput('');
+		setResults([]);
+
+		updateState(false);
+	}
+
+	async function handleSearch(): Promise<any> {
+		const data = await requestByName({ name: searchInput });
+		const aux: Record<any, any>[] = [];
+		data.forEach((result) => {
```

Comment:
Same issue here. Since aux is an array of strings, you can only do `aux: string[]`.